python/splib/geometric/gmsh.py

```diff
@@ -0,0 +1,165 @@
+    ParametricGeometryFile = open(InputFilePath)
+    # Warning: here we are not taking into account that the file could use a large amount of memory
+    FileContents = ParametricGeometryFile.read()
+    
+    # hash the file contents
+    FileAndOptionsHashObj = hashlib.sha256(FileContents)
+    
+    # add the options strings to the hash
+    for i in SortingIdxs:
+        ArgsForHash = OptionsStrings[i] + '=' + str(Values[i]) + ';'
+        FileAndOptionsHashObj.update(ArgsForHash)
+    
+    # Finally, add output file extension to the hash, so that different target files from the same source will be treated differently (e.g. Surface and Volumetric meshes)
+    FileAndOptionsHashObj.update(OutputFileExtension)
+    
+    # Get the hash string and verify if it was previously generated    
+    HashStr = FileAndOptionsHashObj.hexdigest()    
+    # OneShot
+    if OutputFileName != None: 
+        HashFilePath = OutputDir + OutputFileName + OutputFileExtension + '.hash'
+        FilePath = OutputDir + OutputFileName + OutputFileExtension 
+        if os.path.exists(HashFilePath):
+            HashFileRead = open(HashFilePath,'r')
+            OldHashStr = HashFileRead.readline()
+            if OldHashStr == HashStr+'\n':
+                print(FilePath + ': Found a file with an identical hash. Returning from cache.')                
+                return False, FilePath
+        
+        # If hash is different or non-existent write hash (+options) info to file                 
+        HashFile = open(HashFilePath, 'w+')
+        HashFile.write(HashStr+'\n')
+        HashFile.write('# Options:\n')
+                       
+        for i in SortingIdxs:
+                HashFile.write('# ' + OptionsStrings[i]+'='+str(Values[i])+';\n')
+        HashFile.close()
+        return True, FilePath
+    
+    
+    # Persistent
+    else:
+        HashedFileName = HashStr + OutputFileExtension
+        HashedFilePath = OutputDir + HashedFileName
+        if os.path.exists(HashedFilePath):    
+            print(HashedFilePath + ': Found a file with an identical hash. Returning from cache.')                
+            return False, HashedFilePath
+        else:
+            return True, HashedFilePath
+
+    
+def meshFromParametricGeometry(filepath, outputdir='autogen', meshtype='Surface', **kwargs):
+        """generate a tetrahedron mesh from the provided file and store the 
+           result in a vtk file. The filename is returned. 
+           
+           :param str filepath:
+           :param str outputdir:
+           :param float Mesh_CharacteristicLengthFactor:
+           :param float Mesh_CharacteristicLengthMax:
+           :param float Mesh_CharacteristicLengthMin: 
+           :param float View_GeneralizedRaiseZ
+        """ 
+        import gmshpy        
+        import os
+        import numpy as np
+        import locale
+        locale.setlocale(locale.LC_ALL, 'C')
```

Comment:
if this is not already done... use a contextmanager here to restore the initial locale.

Reply:
with mycontextManager.locale("C"):
       ...